src/databricks/labs/ucx/azure/access.py

```diff
@@ -91,6 +103,93 @@ def save_spn_permissions(self) -> str | None:
+        self, policy_definition: str, storage_accounts: list[AzureResource], global_principal: Principal
+    ) -> str:
+        policy_dict = json.loads(policy_definition)
+        tenant_id = self._azurerm.tenant_id()
+        endpoint = f"https://login.microsoftonline.com/{tenant_id}/oauth2/token"
+        for storage in storage_accounts:
+            policy_dict[
+                f"spark_conf.fs.azure.account.oauth2.client.id.{storage.storage_account}.dfs.core.windows.net"
+            ] = self._policy_config(global_principal.client_id)
+            policy_dict[
+                f"spark_conf.fs.azure.account.oauth.provider.type.{storage.storage_account}.dfs.core.windows.net"
+            ] = self._policy_config("org.apache.hadoop.fs.azurebfs.oauth2.ClientCredsTokenProvider")
+            policy_dict[
+                f"spark_conf.fs.azure.account.oauth2.client.endpoint.{storage.storage_account}.dfs.core.windows.net"
+            ] = self._policy_config(endpoint)
+            policy_dict[f"spark_conf.fs.azure.account.auth.type.{storage.storage_account}.dfs.core.windows.net"] = (
+                self._policy_config("OAuth")
+            )
+            if global_principal.secret is not None:
+                policy_dict[
+                    f"spark_conf.fs.azure.account.oauth2.client.secret.{storage.storage_account}.dfs.core.windows.net"
+                ] = self._policy_config(global_principal.secret)
+        return json.dumps(policy_dict)
+
+    @staticmethod
+    def _policy_config(value: str):
+        return {"type": "fixed", "value": value}
+
+    def _update_cluster_policy_with_spn(
+        self, policy_id: str, storage_accounts: list[AzureResource], global_principal: Principal
+    ):
+        try:
+            policy_definition = ""
+            cluster_policy = self._ws.cluster_policies.get(policy_id)
+            # to be updated after doing changes in blueprint
+            # self._installation.save(cluster_policy.as_dict(), filename="policy-backup.json")
+            if cluster_policy.definition is not None:
+                policy_definition = self._update_cluster_policy_definition(
+                    cluster_policy.definition, storage_accounts, global_principal
+                )
+            if cluster_policy.name is not None:
+                self._ws.cluster_policies.edit(policy_id, cluster_policy.name, definition=policy_definition)
+        except NotFound:
+            msg = f"cluster policy {policy_id} not found, please run UCX installation to create UCX cluster policy"
+            raise NotFound(msg) from None
+
+    def create_uber_principal(self):
+        config = self._installation.load(WorkspaceConfig)
+        policy_id = config.policy_id
+        if policy_id is None:
+            msg = "UCX cluster policy not found in config. Please run latest UCX installation to set cluster policy"
+            logger.error(msg)
+            raise ValueError(msg) from None
+        if config.global_spn_id is not None:
```

Comment:
Important: be consistent with naming everywhere. 
```suggestion
        if config.uber_spn_id is not None:
```